src/platform/plugins/shared/discover/public/context_awareness/profile_providers/security/components/row_leading_controls/explore_in_security.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import type { RowControlComponent, RowControlRowProps } from '@kbn/discover-utils';
+import { getFieldValue } from '@kbn/discover-utils';
+import type { ProfileProviderServices } from '../../../profile_provider_services';
+import { getSecurityTimelineRedirectUrl } from '../../utils';
+import { exploreRowActionLabel } from '../../translations';
+
+export interface ExploreInSecurityProps {
+  Control: RowControlComponent;
+  rowProps: RowControlRowProps;
+  services: ProfileProviderServices;
+}
+
+export function ExploreInSecurity({ Control, rowProps, services }: ExploreInSecurityProps) {
+  const hit = rowProps.record;
+  const {
+    application: { getUrlForApp },
+  } = services;
+
+  const timelinesURL = getUrlForApp('securitySolutionUI', {
+    path: 'alerts',
+  });
+
+  const alertURL = useMemo(() => getFieldValue(hit, 'kibana.alert.url') as string, [hit]);
+  const eventKind = useMemo(() => getFieldValue(hit, 'event.kind') as string, [hit]);
+  const isAlert = useMemo(() => eventKind === 'signal', [eventKind]);
+  const eventId = useMemo(() => getFieldValue(hit, '_id') as string, [hit]);
+  const eventURL = useMemo(
+    () =>
+      getSecurityTimelineRedirectUrl({
+        from: getFieldValue(hit, '@timestamp') as string,
+        to: getFieldValue(hit, '@timestamp') as string,
+        eventId: eventId as string,
+        index: getFieldValue(hit, '_index') as string,
+        baseURL: timelinesURL,
+      }),
+    [hit, eventId, timelinesURL]
+  );
+
+  const url = useMemo(() => (isAlert ? alertURL : eventURL), [isAlert, alertURL, eventURL]);
+
+  const onControlClick = useCallback(() => {
+    window.open(url, '_blank');
+  }, [url]);
+
+  const label = useMemo(() => exploreRowActionLabel(isAlert), [isAlert]);
+  return (
+    <Control
```

Comment:
@davismcphee , do you think I can `not` use `Control` component ( provided by Additional Row Controls extension point ) and simply use EUI Button because I want to give option to user if they want to open the link in this tab or a new tab?

Reply:
Sorry for the delay on this, I missed the ping. We added the `Control` component to make sure row actions are consistent, so ideally we should use it here too. Supporting this seems reasonable though, so how about just updating `RowControlComponent` in Unified Data Table to support it? We could accept an `href` prop and use the `getRouterLinkProps` util to handle it, similar to #211802.

Reply:
Thanks @davismcphee , it is done in https://github.com/elastic/kibana/pull/204756/commits/d580cc7660d826e9bf750c8317039174c15baead.